src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Exadel.CrazyPrice.Common.Models.Request;
+using Exadel.CrazyPrice.Common.Models.Response;
+
+namespace Exadel.CrazyPrice.Common.Interfaces
+{
+    public interface ICrazyPriceRepository
+    {        
+
+        /// <summary>
+        /// Gets all Discount Programs.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<DiscountResponse>> GetAllDiscountProgramsAsync();
+                
+        /// <summary>
+        /// Gets a Discount Program by id.
+        /// </summary>
+        /// <param name="id">The id Discount Program.</param>
+        /// <returns></returns>
+        Task<DiscountResponse> GetDiscountProgramAsync(string id);
+
+        /// <summary>
+        /// Gets a Discount for Moderator Program by id Discount Program.
+        /// </summary>
+        /// <param name="id">The id Discount Program.</param>
+        /// <returns></returns>
+        Task<UpsertDiscountRequest> GetDiscountProgramForModeratorAsync(string id);
```

Comment:
Method names strange - does it Get from database or request for upsert

Reply:
[Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/IDiscountRepository.cs)